### PR TITLE
FIX: mass error calculation multiplier AMS

### PR DESCRIPTION
**Pull request by oliveralka**
# Description

Issues with a too high mass error for compounds with a multiplier higher than 1 (Dimer, Trimer).
e.g. 2M+H+

fixes https://github.com/OpenMS/OpenMS/issues/5175

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

File: src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp [hunk 1]
```diff
@@ -116,6 +116,11 @@ namespace OpenMS
     return name_;
   }
 
+  int AdductInfo::getMolMultiplier() const
+  {
+    return mol_multiplier_;
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:121]: initialized to 1 ?

Comment (oliveralka) [reply src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:121]: As far as I have seen it is always initialized to 1 in the `parseAdductString` function.
https://github.com/OpenMS/OpenMS/blob/e10e86f6bccb089bbe422885891d64b179ab1283/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp#L129
https://github.com/OpenMS/OpenMS/blob/e10e86f6bccb089bbe422885891d64b179ab1283/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp#L207

Comment (timosachsenberg) [reply src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:121]: yes but there are other ways to create an AdductInfo object. Please also make sure that methods that set the mol_multiplier_ check for 0 as this is invalid e.g., https://github.com/OpenMS/OpenMS/blob/e10e86f6bccb089bbe422885891d64b179ab1283/src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp#L52

Comment (oliveralka) [reply src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:121]: Should be sufficient to throw an error in the constructor if it is 0, I guess

```
    if (mol_multiplier_ == 0)
    {
      throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Molecule multiplier of 0 is not allowed for an adduct (" + ef_.toString() + ")");
    }
```

File: src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp [hunk 2]
```diff
@@ -116,6 +120,11 @@ namespace OpenMS
     return name_;
   }
 
+  int AdductInfo::getMolMultiplier() const
```
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:123]: UInt instead of int?

Comment (oliveralka) [reply src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp:123]: Yes, that would be better.

**Review by timosachsenberg — APPROVED** [review]

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Comment by oliveralka** [comment]
rebuild jenkins

**Review by FabianAicheler — APPROVED** [review]
Looks fine to me

**Pull request merged**